### PR TITLE
More accurate autolink warnings line

### PR DESCRIPTION
**Pull request by wojtekmach**
Demo:

image

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -34,7 +34,9 @@ defmodule ExDoc.Mixfile do
 
   defp deps do
     [
-      {:earmark_parser, "~> 1.4.31"},
+      # TODO
+      {:earmark_parser, "~> 1.4.31",
+       github: "wojtekmach/earmark_parser", branch: "wm-codespan-line"},
```
Comment (wojtekmach) [inline mix.exs:39]: https://github.com/RobertDober/earmark_parser/pull/140

**Comment by garazdawi** [comment]
When I saw your other PRs regarding warnings I was hoping the change in this PR was coming soon :) looks great! Now we just need the various LSPs to warn for this as well :)

**Review by josevalim — APPROVED** [review]
Absolutely beautiful.

**Comment by garazdawi** [comment]
Do you know of any way to fix this for Erlang as well? `:elixir_error.emit_diagnostic/5` seems to get

```
[{elixir_compiler,'__FILE__',1,
                  [{file,"/home/eluklar/git/otp3/lib/stdlib/src/gen_statem.erl"},
                   {line,1180}]}]
```

as the stacktrace, instead of something that it can convert to a nice warning message.

**Comment by josevalim** [comment]
Elixir allows you to compile code outside of modules, but it does so by compiling the code inside a fake module with a fake `__FILE__` function. So for a warning like that to appear, you need to be compiling code somewhere. Do you know where that could be?

Before printing stacktraces, Elixir explicitly matches on the `:elixir_compiler.__FILE__/1` and knows it should show only the file info.

**Comment by garazdawi** [comment]
I managed to get a bit further by setting `:module` and `:function` in IO.warn. Now the warning I get looks like this:

```
warning: documentation references function "erlang:foo/1" but it is undefined or private
  (stdlib 5.1.1) .../lib/stdlib/src/gen_statem.erl:1181: :gen_statem.callback_mode/0
```

Still need to figure out how to get the code snippet printed, or is that something that mix does?

**Comment by wojtekmach** [comment]
printing snippet is an Elixir 1.16 feature!

**Comment by garazdawi** [comment]
Aha!

**Comment by garazdawi** [comment]
```
      warning: documentation references function "erlang:throws/1" but it is undefined or private
      │
 1181 │ `erlang:throws/1`
      │ ~~~~~~~~~~~~~~~~~
      │
      └─ /home/eluklar/git/otp3/lib/stdlib/src/gen_statem.erl:1181: :gen_statem (module)

      warning: documentation references type "t:callback_mode/0" but it is undefined or private
      │
 1338 │ -doc """
      │ ~~~~~~~~
      │
      └─ /home/eluklar/git/otp3/lib/stdlib/src/gen_statem.erl:1338: :gen_statem (module)
```

with an upgraded Elixir it works great for regular links. Custom links don't have the line number info though so they look a bit odd :) Looking forward to the same info being added about `<a href>`! :D

I'll submit a PR with the changes I had to do to make Erlang work.

**Pull request merged**